### DISCUSSION TITLE
Support root-path

### DISCUSSION THREAD

**Discussion by dema-debug** [fastapi/fastapi#9163] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI(docs_url=None, root_path='aaa')

@app.get("/docs", include_in_schema=False)
async def swagger_ui_html():
    return get_swagger_ui_html(
        openapi_url="/openapi.json",
        title="ABC",
        swagger_favicon_url="/static/favicon.png"
    )
```

### Description

when I set root_path in 'app = FastAPI()',I fount it worked but not totally worked. 
First, all the api_url in swagger had picking root_path, but actual api request in brower still should not contains  root_path, and openapi.json also should not contains.

### Operating System

Linux, Windows

### FastAPI Version

0.79.0

### Python Version

3.9.13

### Additional Context

when I set root_path in 'app = FastAPI()',I fount it worked but not totally worked. 
First, all the api_url in swagger had picking root_path, but actual api request in brower still should not contains  root_path, and openapi.json also should not contains.
Maybe I did not set completely, please help~     QAQ

**Comment by Kludex** [comment, marked as answer]
Related issues with `root-path` are fixed in the latest FastAPI and Uvicorn releases.
